src/Appwrite/Platform/Workers/Migrations.php

```diff
@@ -337,26 +379,28 @@ protected function processMigration(Document $project, Document $migration, Log
                 $errorMessages = [];
                 foreach ($sourceErrors as $error) {
                     /** @var MigrationException $error */
-                    $errorMessages[] = "Error occurred while fetching '{$error->getResourceGroup()}:{$error->getResourceId()}' from source with message '{$error->getMessage()}'";
+                    $errorMessages[] = "Error occurred while fetching '{$error->getResourceName()}:{$error->getResourceId()}' from source with message '{$error->getMessage()}'";
                 }
                 foreach ($destinationErrors as $error) {
                     /** @var MigrationException $error */
-                    $errorMessages[] = "Error occurred while pushing '{$error->getResourceGroup()}:{$error->getResourceId()}' to destination with message '{$error->getMessage()}'";
+                    $errorMessages[] = "Error occurred while pushing '{$error->getResourceName()}:{$error->getResourceId()}' to destination with message '{$error->getMessage()}'";
                 }
 
-                $migrationDocument->setAttribute('errors', $errorMessages);
+                $migration->setAttribute('errors', $errorMessages);
                 $log->addTag('migrationErrors', json_encode($errorMessages));
             }
         } finally {
-            if ($tempAPIKey) {
+            if (! $tempAPIKey->isEmpty()) {
                 $this->removeAPIKey($tempAPIKey);
             }
-            if ($migrationDocument) {
-                $this->updateMigrationDocument($migrationDocument, $projectDocument);
 
-                if ($migrationDocument->getAttribute('status', '') == 'failed') {
-                    throw new Exception("Migration failed");
-                }
+            $this->updateMigrationDocument($migration, $projectDocument);
+
+            if ($migration->getAttribute('status', '') == 'failed') {
+                $destination->error();
+                $source->error();
+
+                throw new Exception('Migration failed');
```

Comment:
Can we also report the reason for the "Migration Failed" error? Maybe convert the error message to a string and attach it to the exception?

Reply:
The errors of sources and destinations are appended, not always have a single error, sometimes the errors list can be very big. I guess we can always make the errors mechanism better in Migrations.php

Reply:
okay 👍